modules/transformer-batch/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/transformer/batch/ShredJob.scala

```diff
@@ -128,12 +130,9 @@ class ShredJob[T](@transient val spark: SparkSession,
 
```

Comment:
We have 3 types of deduplication (please correct me if I'm wrong!):

1) in-batch natural, same `event_id` and `fingerprint`
2) in-batch synthetic, same `event_id` different `fingerprint`
3) cross-batch natural, same `event_id` and `fingerprint`

After this change: 

* '3)' is not allowed if '1)' is disabled
* '2)' is allowed if '1)' is disabled

So my main question is about the second point - should we allow synthetic if natural is disabled?

Reply:
That is a good catch! I rechecked the implementation of synthetic deduplication. Seems like it assumes that natural deduplication is already done. If natural deduplication isn't done, it will process natural duplicate items as synthetic duplicate which is wrong. I will add additional check to make sure in-batch natural deduplication is enabled when synthetic deduplication is enabled.